doc: nrf_modem: updates to delta DFU (#610)
nrfconnect/sdk-nrfxlib: nrf_modem/doc/delta_dfu.rst

Upon success, the operation is scheduled and the modem updates its firmware the next time the application boots the modem by calling :c:func:`nrf_modem_init`.

.. note::
   The return value of :c:func:`nrf_modem_delta_dfu_update` only indicates whether the modem has scheduled the update and does not represent the result of the actual update operation.

> .. figure:: images/delta_dfu_update.svg
>    :alt: Modem delta DFU update flowchart

Reviewer: ```suggestion
.. figure:: images/delta_dfu_update.svg
   :alt: Modem delta DFU update flowchart
   
   Modem delta DFU update flowchart
```
This adds the caption.

The same firmware delta that is used to update to a new firmware version can be used to roll back to the previous firmware version.
Thus, the application need not download the previous modem firmware version but can perform the rollback entirely in an offline mode, without registering to the network.

.. important::
   The rollback functionality is only available as long as the application does not erase the contents of the scratch area by calling :c:func:`nrf_modem_delta_dfu_erase`.
   Once the application has called :c:func:`nrf_modem_delta_dfu_erase` it can no longer roll back to a previous modem firmware version using the delta DFU API.

To roll back to the previous modem firmware version, the application calls :c:func:`nrf_modem_delta_dfu_rollback`.
Upon success, the operation is scheduled and the modem performs a rollback to the previous modem firmware when the application boots it by calling :c:func:`nrf_modem_init`.
The return value of :c:func:`nrf_modem_delta_dfu_rollback` only indicates whether the modem has scheduled the rollback and does not represent the result of the actual rollback operation.

The figure below shows the flow chart for the rollback operation.

> .. figure:: images/delta_dfu_rollback.svg
>    :alt: Modem delta DFU rollback flowchart

Reviewer: ```suggestion
.. figure:: images/delta_dfu_rollback.svg
   :alt: Modem delta DFU rollback flowchart
   
   Modem delta DFU rollback flowchart
```
Adding the caption.

Checking the result of the rollback
===================================

:c:func:`nrf_modem_init` will return one of the following values:

* ``MODEM_DFU_RESULT_OK`` - The rollback is successful. The modem will run the previous firmware the next time it boots.
* ``MODEM_DFU_RESULT_AUTH_ERROR`` - The rollback did not take place. The modem will run the same firmware the next time it boots.
* ``MODEM_DFU_RESULT_UUID_ERROR`` - The rollback did not take place. The modem will run the same firmware the next time it boots.
* ``MODEM_DFU_RESULT_INTERNAL_ERROR`` - The modem encountered an internal error while executing the rollback, and it will not boot to prevent executing unintended operations. For subsequent programming, the modem can only be programmed through the :ref:`nrf_modem_full_dfu_api`.
* ``MODEM_DFU_RESULT_HARDWARE_ERROR`` - The modem encountered a hardware error while executing the rollback, and it will not boot to prevent executing unintended operations. For subsequent programming, the modem can only be programmed through the :ref:`nrf_modem_full_dfu_api`.

Reinitializing the modem to run the firmware
============================================

> Regardless of the value returned by :c:func:`nrf_modem_init`, the application must reinitialize the modem by reinitializing the Modem library in order to run the modem firmware.

Reviewer: ```suggestion
Regardless of the value returned by :c:func:`nrf_modem_init`, the application must reinitialize the modem by reinitializing the :ref:`nrf_modem` to run the modem firmware.
```

The application can reinitialize the Modem library by calling :c:func:`nrf_modem_shutdown` followed by :c:func:`nrf_modem_init`.
If the rollback is successful, the modem runs the previous modem firmware.

> The application can verify that by reading the modem firmware UUID or reading the ``AT+CGMR`` command response.

Reviewer: ```suggestion
The application can verify that by reading the modem firmware UUID or the ``AT+CGMR`` command response.
```
